ext/drx/drx.c

```diff
@@ -2358,7 +2416,22 @@ drx_expand_scatter_gather(void *drcontext, instrlist_t *bb, OUT bool *expanded)
             scratch_xmm != reg_resize_to_opsz(sg_info.gather_dst_reg, OPSZ_16))
             break;
     }
-    /* FIXME i#2985: spill scratch_xmm using a future drreg extension for simd. */
+    /* Spill the scratch xmm.
+     * TODO i#3844: drreg does not support spilling xmm regs yet, so we do it ourselves.
+     * When that support is available, replace the following with the required drreg API
+     * calls.
```

Comment:
What about restore_state?  I don't see any change here: are thread redirects disallowed while scratch_xmm is clobbered?

Reply:
Added.